Lib/_pydecimal.py

```diff
@@ -500,7 +504,12 @@ def sin(x):
     >>> print(getcontext().prec)
     28
     """
-    if ctx is None: ctx = getcontext()
+    if ctx is None:
+        ctx = getcontext()
```

Comment:
It looks like `getcontext()` doesn't copy the previous context - that's done in '_ContextManager` - so by editing here, the changes would persist after the context manager has exited?